bin/common.bash

```diff
@@ -399,12 +399,17 @@ validate_sops_config() {
         exit 1
     fi
 
-    # Force all pgp entries to be a string and to contain "null" if unset in a creation rule
-    fingerprints=$(yq4 'with(.creation_rules[]; .pgp = (.pgp // "null" | to_yaml style="")) | [.creation_rules[].pgp] | join(",")' "${sops_config}")
-    if ! [[ "${fingerprints}" =~ ^[A-Z0-9,' ']+$ ]]; then
+    # Compares the keyring with pgp fingerprints to see if pgp has anything keyring does not have.
+    keyring=$(gpg --list-keys  | grep pub -A 1 | grep -v pub | sed -n 'n;p' | xargs)
+    creation_pgp=$(yq4 '[.creation_rules[].pgp // null | split(",") | .[]] | unique | join(" ")' "${sops_config}")
+    comma_search=$(yq4 '.creation_rules[] | select(.pgp == "*,")' "${sops_config}")
```

Comment:
What is the comma search for? :thinking:

Reply:
If there is a comma at the end of a pgp it will not re encrypt the secrets.yaml so this way those errors will also be caught.
- - pgp: workingexamplekey,       So this won't re encrypt.
- - pgp: workingexamplekey        This will.

Reply:
I see.